core-library/src/main/java/org/silverpeas/core/index/search/SimpleSearchEngine.java

```diff
@@ -151,19 +152,19 @@ private boolean isSpellingNeeded(List<MatchingIndexEntry> results) {
     }
 
     for (MatchingIndexEntry result : matchingIndexEntries) {
-      if (!isMatchingIndexEntryAvailable(result, userId, authorization, allowedComponentIds)) {
+      if (!isMatchingIndexEntryAvailable(result, userId, authorizations, allowedComponentIds)) {
         continue;
       }
       results.add(result);
     }
 
-    authorization.disableCache();
+    disableCache(authorizations);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "disableCache()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS2259)